evm/src/cpu/kernel/tests/bignum/mod.rs

```diff
@@ -0,0 +1,394 @@
+use std::cmp::Ordering;
+use std::fs::File;
+use std::io::{BufRead, BufReader};
+use std::path::PathBuf;
+
+use anyhow::Result;
+use ethereum_types::U256;
+use itertools::Itertools;
+use num::{BigUint, One, Zero};
+use num_bigint::RandBigInt;
+use plonky2_util::ceil_div_usize;
+use rand::Rng;
+
+use crate::cpu::kernel::aggregator::KERNEL;
+use crate::cpu::kernel::interpreter::Interpreter;
+use crate::util::{biguint_to_mem_vec, mem_vec_to_biguint};
+
+const BIGNUM_LIMB_BITS: usize = 128;
+const MINUS_ONE: U256 = U256::MAX;
+
+const TEST_DATA_BIGNUM_INPUTS: &str = "bignum_inputs";
+const TEST_DATA_U128_INPUTS: &str = "u128_inputs";
+const TEST_DATA_SHR_OUTPUTS: &str = "shr_outputs";
+const TEST_DATA_ISZERO_OUTPUTS: &str = "iszero_outputs";
+const TEST_DATA_CMP_OUTPUTS: &str = "cmp_outputs";
+const TEST_DATA_ADD_OUTPUTS: &str = "add_outputs";
+const TEST_DATA_ADDMUL_OUTPUTS: &str = "addmul_outputs";
+const TEST_DATA_MUL_OUTPUTS: &str = "mul_outputs";
+
+const BIT_SIZES_TO_TEST: [usize; 4] = [128, 256, 512, 1000];
```

Comment:
You should probably include borderline bitsizes as well, like 0, 1, 2, 127, 129, 255, 257, 1023, 1024, 1025, and maybe a much bigger one like 31415. You'll need to adjust `max_bignum` and `test_iszero_bignum_all` (and maybe some others) to accommodate bitsize 0.